### DISCUSSION TITLE
New MOOSE update does not recognise files

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#25738] [Q&A General]
Hello, I have reinstalled MOOSE and I am receiving some errors when running old simulations that were working before the update. The procedure that I was following before was creating the necessary files that I needed and I was adding them to the "materials" directory in Blackbear for "src" and "include" directories. Then I was running "make" in Blackbear to include these new files for my simulations. I know it is not the most common way to do it but it was working fine and it was the easiest way because they were only 2 files.
I have tried to do the same now but it seems like the functions from the added files in Blackbear are not recognise and I am receiving errors such as:

*** ERROR ***
The following error occurred in the object "k_func", of type "PiecewiseLinear".

A parameter was passed for an option using data from a CSV file but the 'data_file' parameter has not been set. This is not allowed

Is it because I am not including correctly other modules? I have tried to change the Makefile and set it to include all modules, but it is not working. I am running the simulation as:

mpirun -np 20 path/blackbear/blackbear-opt -i MaterialDeposition.i

Thank you in advance.

**Comment by GiudGiud** [comment]
Hello

I modified these parameters last month.
MOOSE was silently ignoring  piecewise tabular (linear/constant) function parameters that were set and not used. You will need to adapt your app. Are you already using the latest BlackBear version?

My guess is that you set the data_type parameter while using input file xy data 
data_type is only used for the CSV input

Guillaume

**Reply by fervaldies** [reply]
Thank you! I am indeed using the latest BlackBear version, I have changed some things and now it is working, but I have another error regarding the type for a phase-field simulation. I am trying to run a phase-field simulation but it tells me that the PolycrystalSolidKernel needs a type (it is a new action), but when I am using the normal Polycrystalkernel included in MOOSE I do not see that error message. Do you know why this could be?

*** ERROR ***
path/solid.i:19: missing required parameter 'Kernels/PolycrystalSolidKernel/type'
        Doc String: "A string representing the Moose Object that will be built by this Action"

**Reply by GiudGiud** [reply]
Is it a MooseObjectAction ? MooseObjectAction use the type required parameter

**Reply by GiudGiud** [reply]
> *** ERROR ***
The following error occurred in the object "k_func", of type "PiecewiseLinear".

> A parameter was passed for an option using data from a CSV file but the 'data_file' parameter has not been set. This is not allowed

I improved the error message for the next person who ll hit this. Thanks for reporting this problem.
